### DISCUSSION TITLE
Getting Associated Patient from PATIENT_ADDRESS and PATIENT_CONTACT_POINT Events

### DISCUSSION THREAD

**Discussion by tamalhu** [canvas-medical/canvas-plugins#259] [Q&A]
Hi Canvas team,

I'm currently working on a plugin and I noticed that the `target` for the `PATIENT_ADDRESS` and `PATIENT_CONTACT_POINT` events is the ID of the address / contact entity respectively. I'd like to grab the patient that these entities belong to, but these events don't seem to supply the patient ID anywhere.

Is there a way to get the patient ID from these events? If not, I think the `context` dictionary would be a great place to include this value.

I've also heard that the upcoming changes here https://github.com/canvas-medical/canvas-plugins/issues/188 may help with my issue since it sounds like they'll introduce the `Address` and `Telecom` models to the SDK data module, which I guess might support grabbing their associated `Patient`?

**Comment by kristenoneill** [comment]
@tamalhu - Thanks for flagging. We do have this work prioritized.  #209 accounts for adding the patient ID as context to the events you listed. If you have related questions or concerns, please don't hesitate to continue the discussion on #209.

**Comment by rmagier1** [comment]
These events will have a `self.context` that will contain the patient identifier 
<img width="871" height="373" alt="Screenshot 2026-02-05 at 4 53 32 PM" src="https://github.com/user-attachments/assets/1f355287-6d92-43b7-b145-06c808ce18d7" />
